src/api/app/controllers/webui/users/notifications_controller.rb

```diff
@@ -19,21 +55,12 @@ def index
       @notifications = Notification.with_notifiable.unread
                                    .where(notifiable_type: 'BsRequest')
                                    .for_subscribed_user(User.session)
-    else
+    when 'unread'
       @notifications = Notification.with_notifiable.unread
                                    .for_subscribed_user(User.session)
-    end
-  end
-
-  def update
-    notification = User.session.notifications.find(params[:id])
-    authorize notification, policy_class: NotificationPolicy
-
-    if notification.toggle(:delivered).save
-      flash[:success] = "Successfully marked the notification as #{notification.unread? ? 'unread' : 'read'}"
     else
```

Comment:
The same than before here, this method is getting data and the `flash` and the `redirect_to` are too much responsibilities. I would move that out

Reply:
This code was also moved.